src/ape_cache/query.py

```diff
@@ -0,0 +1,336 @@
+from pathlib import Path
+from typing import Any, List, Optional
+
+import pandas as pd
+import sqlalchemy.exc
+from sqlalchemy import create_engine  # type: ignore
+from sqlalchemy.sql import text  # type: ignore
+
+from ape.api import QueryAPI, QueryType
+from ape.api.networks import LOCAL_NETWORK_NAME
+from ape.api.query import BlockQuery, BlockTransactionQuery, ContractEventQuery
+from ape.exceptions import QueryEngineError
+from ape.logging import logger
+from ape.utils import singledispatchmethod  # type: ignore
+
+from . import models
+
+TABLE_NAME = {
+    BlockQuery: "blocks",
+    BlockTransactionQuery: "transactions",
+    ContractEventQuery: "contract_events",
+}
+
+
+class CacheQueryProvider(QueryAPI):
+    """
+    Default implementation of the ape.api.query.QueryAPI
+    Allows for the query of blockchain data using connected provider
+    """
+
+    @property
+    def database_file(self) -> Path:
```

Comment:
I think have this be `Optional[Path]` so that we can effectively "bypass" using the cache if the DB has been purged, requiring `ape cache init` first before activating the cache